python/examples/sample-selection.py

```diff
@@ -0,0 +1,113 @@
+import sys
+
+import chemfiles
+import numpy as np
+from equistore import Labels
+
+from rascaline import SoapPowerSpectrum
+
+
+with chemfiles.Trajectory(sys.argv[1]) as trajectory:
+    frames = [f for f in trajectory]
+
+HYPER_PARAMETERS = {
+    "cutoff": 5.0,
+    "max_radial": 6,
+    "max_angular": 4,
+    "atomic_gaussian_width": 0.3,
+    "center_atom_weight": 1.0,
+    "radial_basis": {
+        "Gto": {},
+    },
+    "cutoff_function": {
+        "ShiftedCosine": {"width": 0.5},
+    },
+}
+
+calculator = SoapPowerSpectrum(**HYPER_PARAMETERS)
+
+descriptor = calculator.compute(frames)
+
+# The selections for sample can be a set of `Labels`, in which case
+# the names of the labels must be a subset of the names of the
+# samples produced by the calculator. You can see the default set
+# of names with:
+print("samples names:", descriptor.sample_names)
+
+# Now we can use these names for selection.
+# In this case, only samples  matching the labels in this selection
+# will be used
+selection = Labels(
+    names=["structure"],
+    values=np.array([[0], [2], [3]]),
+)
+
+reference_descriptor = calculator.compute(frames, selected_samples=selection)
+
+# We can compare obtained features matrix with the matrix, which we
+# will get by direct selection of atomic environments from the
+# full matrix.
+
+# obtain indices of corresponding atoms
+indices = [[i for i in range(len(frames[0].atoms))]]
+indices.append(
+    [
+        len(frames[0].atoms) + len(frames[1].atoms) + i
+        for i in range(len(frames[2].atoms))
+    ]
+)
+indices.append(
+    [
+        len(frames[0].atoms) + len(frames[1].atoms) + len(frames[2].atoms) + i
+        for i in range(len(frames[2].atoms))
+    ]
+)
+indices = np.array(indices).reshape(
+    -1,
+)
+
+reference_descriptor.keys_to_samples("species_center")
+reference_descriptor.keys_to_properties(["species_neighbor_1", "species_neighbor_2"])
+descriptor.keys_to_samples("species_center")
+descriptor.keys_to_properties(["species_neighbor_1", "species_neighbor_2"])
+
+reference_features = reference_descriptor.block(0).values
+features_direct = descriptor.block(0).values
+features_direct = features_direct[indices]
+print("Are they are equal?", np.array_equal(reference_features, features_direct))
```

Comment:
This seems to be a bit convoluted to show that we have something. How about printing the samples for some of the blocks, or grouping the blocks and then doing something like:

```
reference_descriptor.keys_to_samples("species_center")
reference_descriptor.keys_to_properties(["species_neighbor_1", "species_neighbor_2"])

samples = reference_descriptor.block().samples
print(f"we have the following structures: {np.unique(samples['structure'])}")
```

Reply:
True yes, it's much easier that way